### PR TITLE
All reworks in options structures

### PR DESCRIPTION
**Pull request by fernando457829**
### It includes reworks in:
- Connections
- Listeners
- Loaders

### CODE REVIEW

**Comment by fernando457829** [comment]
@pedrofracassi done

**Comment by fernando457829** [comment]
OH FUCK, I CLOSED IT UNINTENTIONALLY AAAAAAAAAA i'm idiot

File: src/apis/SteamLadder.js [hunk 1]
```diff
@@ -3,7 +3,10 @@ const SteamLadder = require('steamladder')
 
 module.exports = class SteamLadderAPI extends APIWrapper {
   constructor () {
-    super()
+    super({
+      name: 'steamladder',
+      envVars: ['STEAM_LADDER_API_KEY']
+    })
     this.name = 'steamladder'
```
Comment (metehus) [inline src/apis/SteamLadder.js]: should this be here?

Comment (pedrofracassi) [reply src/apis/SteamLadder.js]: Don't think so

File: src/apis/Spotify.js [hunk 2]
```diff
@@ -7,9 +7,11 @@ const API_URL = 'https://api.spotify.com/v1'
...
   constructor () {
-    super()
-    this.name = 'spotify'
-    this.envVars = ['SPOTIFY_CLIENT_ID', 'SPOTIFY_CLIENT_SECRET']
+    super({
+      name: 'spotify',
+      envVars: ['SPOTIFY_CLIENT_ID', 'SPOTIFY_CLIENT_SECRET']
+    })
+
     this.token = null
```
Comment (pedrofracassi) [inline src/apis/Spotify.js]: Is this supposed to be here?

File: src/apis/LeagueOfLegends.js [hunk 3]
```diff
@@ -5,8 +5,10 @@ const API_URL = 'https://ddragon.leagueoflegends.com'
 
 module.exports = class LeagueOfLegends extends APIWrapper {
   constructor () {
-    super()
-    this.name = 'lol'
+    super({
+      name: 'lol'
+    })
+
     this.version = null
```
Comment (pedrofracassi) [inline src/apis/LeagueOfLegends.js]: Is this supposed to be here?

File: src/apis/Spotify.js (hunk 2 above)
Comment (fernando457829) [reply src/apis/Spotify.js]: I'm forgot2

File: src/apis/LeagueOfLegends.js (hunk 3 above)
Comment (fernando457829) [reply src/apis/LeagueOfLegends.js]: I'm forgot1

File: src/apis/Spotify.js (hunk 2 above)
Comment (fernando457829) [reply src/apis/Spotify.js]: Ohh, my fork is outdated, spotify connection wasn't there

**Comment by pedrofracassi** [comment]
@fernando457829 	fix it fix it fix it

**Comment by pedrofracassi** [comment]
Is that everything?

**Pull request closed without merging**